lib/events/ViewEventListener.php

```diff
@@ -15,9 +16,6 @@ class ViewEventListener {
 	 */
 	public function onKernelView(GetResponseForControllerResultEvent $event) {
 		$value = $event->getControllerResult();
-
-		if ($value instanceof View) {
-			$event->setResponse(new Response(view_to_string($value)));
-		}
+		$event->setResponse(as_response($value->toResponse()));
```

Comment:
Misschien wel een if `ToResponse` neerzetten, dan is het duidelijk wanneer deze listener iets doet.

Reply:
Je bedoelt hier de inhoud van as_response reproduceren in plaats van die functie aanroepen?

Reply:
Ja, op de andere plek waar as_response gebruikt wordt weet je toch al zeker dat je een `ToResponse` krijgt.

Reply:
Denk wel beter om de `$event->setResponse` alleen uit te voeren als er iets gebeurt. Deze methode kan maar in één listener uitgevoerd worden.

De exceptie hier is overbodig, want Symfony gooit zelf ook al een (behulpzamere) exceptie als er geen Response wordt gereturned uit een controller (en er is geen listener die dit oplapt).

![](https://i.imgur.com/y3J3poh.png)